BraveAlerterConfigurator.js

```diff
@@ -0,0 +1,111 @@
+// In-house dependencies
+const { BraveAlerter, AlertSession, ALERT_STATE, helpers } = require('brave-alert-lib')
+const db = require('./db/db.js')
+const redis = require('./db/redis.js')
+const ioSockets = require('./ioSockets.js')
+
+const incidentTypes = [
+    'No One Inside',
+    'Person responded',
+    'Overdose',
+    'None of the above'
+]
+
+const incidentTypeKeys = ['1', '2', '3', '4']
+
+class BraveAlerterConfigurator {
+    constructor(startTimes) {
+        this.startTimes = startTimes
+    }
+
+    createBraveAlerter() {
+        return new BraveAlerter(
+            this.getAlertSession.bind(this),
+            this.getAlertSessionByPhoneNumber.bind(this),
+            this.alertSessionChangedCallback.bind(this),
+            false,
+            this.getReturnMessage.bind(this),
+        )
+    }
+
+    async buildAlertSession(session) {
+        let locationData = await db.getLocationData(session.locationid)
+
+        let alertSession = new AlertSession(
+            session.sessionid,
+            session.chatbot_state,
+            session.incidenttype,
+            undefined,
+            `An alert to check on the washroom at ${locationData.location_human} was not responded to. Please check on it`,
+            locationData.phonenumber,
+            incidentTypeKeys,
+            incidentTypes,
+        )
+
+        return alertSession
+    }
+
+    async getAlertSession(sessionId) {
+        let session = await db.getSessionWithSessionId(sessionId)
+
+        return await this.buildAlertSession(session)
+    }
+
+    async getAlertSessionByPhoneNumber(phoneNumber) {
+        let session = await db.getMostRecentSessionPhone(phoneNumber)
+
+        return await this.buildAlertSession(session)
+    }
+
+    async alertSessionChangedCallback(alertSession) {
+        const incidentType = incidentTypes[incidentTypeKeys.indexOf(alertSession.incidentCategoryKey)]
+        await db.saveChatbotSession(alertSession.alertState, incidentType, alertSession.sessionId)
+
+        const session = await db.getSessionWithSessionId(alertSession.sessionId)
+        const locationId = session.locationid
+
+        if (alertSession.alertState === ALERT_STATE.COMPLETED) {
```

Comment:
Need to check and test all this to make sure that all the Redis and ioSockets part works correctly.